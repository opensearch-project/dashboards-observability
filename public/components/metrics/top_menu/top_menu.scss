/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

.resolutionSelect {
  max-width: 300px;
}

.metrics-search-bar-input {
  max-width: 500px;
}

.metrics-search-bar-datepicker {
  width: 500px;
}

.resolutionSelectOption {
  line-height: inherit;
}
