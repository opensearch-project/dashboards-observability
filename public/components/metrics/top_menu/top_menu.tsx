/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  EuiButtonIcon,
  EuiCompressedFieldText,
  EuiCompressedSelect,
  EuiCompressedSuperDatePicker,
  EuiFlexGroup,
  EuiFlexItem,
} from '@elastic/eui';
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { resolutionOptions } from '../../../../common/constants/metrics';
import { uiSettingsService } from '../../../../common/utils';
import { dateSpanFilterSelector, setDateSpan } from '../redux/slices/metrics_slice';
import { MetricsExport } from './metrics_export';
import './top_menu.scss';

export const TopMenu = () => {
  // Redux tools
  const dispatch = useDispatch();
  const dateSpanFilter = useSelector(dateSpanFilterSelector);

  return (
    dateSpanFilter && (
      <>
        <EuiFlexGroup gutterSize="s" justifyContent="spaceBetween" alignItems="center">
          <EuiFlexItem grow={false}>
            <div className="resolutionSelect">
              <EuiCompressedFieldText
                className="resolutionSelectText"
                prepend="Span Interval"
                value={dateSpanFilter.span}
                isInvalid={dateSpanFilter.span < 1}
                onChange={(e) => dispatch(setDateSpan({ span: e.target.value }))}
                data-test-subj="metrics__spanValue"
                append={
                  <EuiCompressedSelect
                    className="resolutionSelectOption"
                    options={resolutionOptions}
                    value={dateSpanFilter.resolution}
                    onChange={(e) => dispatch(setDateSpan({ resolution: e.target.value }))}
                    aria-label="resolutionSelect"
                    data-test-subj="metrics__spanResolutionSelect"
                  />
                }
                aria-label="resolutionField"
              />
            </div>
          </EuiFlexItem>
          <EuiFlexItem grow={false}>
            <EuiFlexGroup gutterSize="s">
              <EuiFlexItem grow={false}>
                <EuiCompressedSuperDatePicker
                  dateFormat={uiSettingsService.get('dateFormat')}
                  start={dateSpanFilter.start}
                  end={dateSpanFilter.end}
                  onTimeChange={(dateSpan) => dispatch(setDateSpan(dateSpan))}
                  recentlyUsedRanges={dateSpanFilter.recentlyUsedRanges}
                  showUpdateButton={false}
                />
              </EuiFlexItem>
              <EuiFlexItem grow={false}>
                <EuiButtonIcon
                  iconType="refresh"
                  aria-label="Refresh"
                  display="base"
                  onClick={() => dispatch(setDateSpan(dateSpanFilter))}
                  size="s"
                  data-test-subj="superDatePickerApplyTimeButton"
                  data-click-metric-element="metrics.refresh_button"
                />
              </EuiFlexItem>
              <EuiFlexItem grow={false}>
                <MetricsExport />
              </EuiFlexItem>
            </EuiFlexGroup>
          </EuiFlexItem>
        </EuiFlexGroup>
      </>
    )
  );
};
