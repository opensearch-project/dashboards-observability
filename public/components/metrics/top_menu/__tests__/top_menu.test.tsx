/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { waitFor } from '@testing-library/react';
import { createStore } from '@reduxjs/toolkit';
import { rootReducer } from '../../../../framework/redux/reducers';
import { Provider } from 'react-redux';
import { TopMenu } from '../top_menu';

describe('Metrics Top Menu Component', () => {
  configure({ adapter: new Adapter() });
  const store = createStore(rootReducer);

  it('renders Top Menu Component when enabled', async () => {
    const wrapper = mount(
      <Provider store={store}>
        <TopMenu />
      </Provider>
    );
    wrapper.update();

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
