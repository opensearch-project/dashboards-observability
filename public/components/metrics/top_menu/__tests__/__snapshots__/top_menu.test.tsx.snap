// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Metrics Top Menu Component renders Top Menu Component when disabled in edit mode 1`] = `
Array [
  <div
    className="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
  >
    <div
      className="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <div
        className="resolutionSelect"
      >
        <div
          className="euiFormControlLayout euiFormControlLayout--group"
        >
          <label
            className="euiFormLabel euiFormControlLayout__prepend"
          >
            Span Interval
          </label>
          <div
            className="euiFormControlLayout__childrenWrapper"
          >
            <input
              aria-label="resolutionField"
              className="euiFieldText resolutionSelectText euiFieldText--inGroup"
              data-test-subj="metrics__spanValue"
              disabled={true}
              onChange={[Function]}
              type="text"
              value={1}
            />
          </div>
          <div
            className="euiFormControlLayout"
          >
            <div
              className="euiFormControlLayout__childrenWrapper"
            >
              <select
                aria-label="resolutionSelect"
                className="euiSelect euiFormControlLayout__append resolutionSelectOption"
                data-test-subj="metrics__spanResolutionSelect"
                onChange={[Function]}
                onMouseUp={[Function]}
                value="h"
              >
                <option
                  key="0"
                  value="s"
                >
                  seconds
                </option>
                <option
                  key="1"
                  value="m"
                >
                  minutes
                </option>
                <option
                  key="2"
                  value="h"
                >
                  hours
                </option>
                <option
                  key="3"
                  value="d"
                >
                  days
                </option>
                <option
                  key="4"
                  value="y"
                >
                  years
                </option>
              </select>
              <div
                className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
              >
                <span
                  className="euiFormControlLayoutCustomIcon"
                >
                  <svg
                    aria-hidden={true}
                    className="euiIcon euiIcon--medium euiFormControlLayoutCustomIcon__icon"
                    focusable="false"
                    height={16}
                    role="img"
                    style={null}
                    viewBox="0 0 16 16"
                    width={16}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M13.069 5.157 8.384 9.768a.546.546 0 0 1-.768 0L2.93 5.158a.552.552 0 0 0-.771 0 .53.53 0 0 0 0 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 0 0 0-.76.552.552 0 0 0-.771 0Z"
                      fillRule="non-zero"
                    />
                  </svg>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="euiFlexItem metrics-search-bar-datepicker"
    >
      <div
        className="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--directionRow euiSuperDatePicker__flexWrapper"
      >
        <div
          className="euiFlexItem"
        >
          <div
            className="euiFormControlLayout euiFormControlLayout--group euiFormControlLayout-isDisabled euiSuperDatePicker"
          >
            <div
              className="euiPopover euiPopover--anchorDownLeft"
            >
              <div
                className="euiPopover__anchor euiQuickSelectPopover__anchor"
              >
                <button
                  aria-label="Date quick select"
                  className="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--xSmall euiButtonEmpty-isDisabled euiFormControlLayout__prepend"
                  data-test-subj="superDatePickerToggleQuickMenuButton"
                  disabled={true}
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    className="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                  >
                    <svg
                      aria-hidden={true}
                      className="euiIcon euiIcon--small euiIcon--inherit euiButtonContent__icon"
                      focusable="false"
                      height={16}
                      role="img"
                      style={null}
                      viewBox="0 0 16 16"
                      width={16}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M13.069 5.157 8.384 9.768a.546.546 0 0 1-.768 0L2.93 5.158a.552.552 0 0 0-.771 0 .53.53 0 0 0 0 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 0 0 0-.76.552.552 0 0 0-.771 0Z"
                        fillRule="non-zero"
                      />
                    </svg>
                    <span
                      className="euiButtonEmpty__text euiQuickSelectPopover__buttonText"
                    >
                      <svg
                        aria-hidden={true}
                        className="euiIcon euiIcon--medium"
                        focusable="false"
                        height={16}
                        role="img"
                        style={null}
                        viewBox="0 0 16 16"
                        width={16}
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M14 4v-.994C14 2.45 13.55 2 12.994 2H11v1h-1V2H6v1H5V2H3.006C2.45 2 2 2.45 2 3.006v9.988C2 13.55 2.45 14 3.006 14h9.988C13.55 14 14 13.55 14 12.994V5H2V4h12zm-3-3h1.994C14.102 1 15 1.897 15 3.006v9.988A2.005 2.005 0 0 1 12.994 15H3.006A2.005 2.005 0 0 1 1 12.994V3.006C1 1.898 1.897 1 3.006 1H5V0h1v1h4V0h1v1zM4 7h2v1H4V7zm3 0h2v1H7V7zm3 0h2v1h-2V7zM4 9h2v1H4V9zm3 0h2v1H7V9zm3 0h2v1h-2V9zm-6 2h2v1H4v-1zm3 0h2v1H7v-1zm3 0h2v1h-2v-1z"
                          fillRule="evenodd"
                        />
                      </svg>
                    </span>
                  </span>
                </button>
              </div>
            </div>
            <div
              className="euiFormControlLayout__childrenWrapper"
            >
              <div
                className="euiDatePickerRange euiDatePickerRange--inGroup"
              >
                <button
                  className="euiSuperDatePicker__prettyFormat euiSuperDatePicker__prettyFormat--disabled"
                  data-test-subj="superDatePickerShowDatesButton"
                  disabled={true}
                  onClick={[Function]}
                >
                  Last 1 day
                  <span
                    className="euiSuperDatePicker__prettyFormatLink"
                  >
                    Show dates
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <span
            className="euiToolTipAnchor"
            onKeyUp={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <button
              className="euiButton euiButton--primary euiButton--fill euiButton-isDisabled euiSuperUpdateButton"
              data-test-subj="superDatePickerApplyTimeButton"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "minWidth": undefined,
                }
              }
              type="button"
            >
              <span
                className="euiButtonContent euiButton__content"
              >
                <svg
                  aria-hidden={true}
                  className="euiIcon euiIcon--medium euiIcon--inherit euiButtonContent__icon"
                  focusable="false"
                  height={16}
                  role="img"
                  style={null}
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11.228 2.942a.5.5 0 1 1-.538.842A5 5 0 1 0 13 8a.5.5 0 1 1 1 0 6 6 0 1 1-2.772-5.058ZM14 1.5v3A1.5 1.5 0 0 1 12.5 6h-3a.5.5 0 0 1 0-1h3a.5.5 0 0 0 .5-.5v-3a.5.5 0 1 1 1 0Z"
                  />
                </svg>
                <span
                  className="euiButton__text euiSuperUpdateButton__text"
                >
                  Refresh
                </span>
              </span>
            </button>
          </span>
        </div>
      </div>
    </div>
    <div
      className="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <div
        className="euiPopover euiPopover--anchorDownCenter"
      >
        <div
          className="euiPopover__anchor"
        >
          <button
            className="euiButton euiButton--primary euiButton-isDisabled"
            data-test-subj="metrics__saveManagementPopover"
            disabled={true}
            onClick={[Function]}
            style={
              Object {
                "minWidth": undefined,
              }
            }
            type="button"
          >
            <span
              className="euiButtonContent euiButtonContent--iconRight euiButton__content"
            >
              <svg
                aria-hidden={true}
                className="euiIcon euiIcon--medium euiIcon--inherit euiButtonContent__icon"
                focusable="false"
                height={16}
                role="img"
                style={null}
                viewBox="0 0 16 16"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13.069 5.157 8.384 9.768a.546.546 0 0 1-.768 0L2.93 5.158a.552.552 0 0 0-.771 0 .53.53 0 0 0 0 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 0 0 0-.76.552.552 0 0 0-.771 0Z"
                  fillRule="non-zero"
                />
              </svg>
              <span
                className="euiButton__text"
              >
                Save
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="euiSpacer euiSpacer--m"
  />,
  <div
    className="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
  >
    <div
      className="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <button
        className="euiButton euiButton--danger euiButton--small"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "minWidth": undefined,
          }
        }
        type="button"
      >
        <span
          className="euiButtonContent euiButton__content"
        >
          <svg
            aria-hidden={true}
            className="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
            focusable="false"
            height={16}
            role="img"
            style={null}
            viewBox="0 0 16 16"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          />
          <span
            className="euiButton__text"
          >
            Cancel
          </span>
        </span>
      </button>
    </div>
    <div
      className="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <button
        className="euiButton euiButton--primary euiButton--small"
        data-test-subj="metrics__saveView"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "minWidth": undefined,
          }
        }
        type="button"
      >
        <span
          className="euiButtonContent euiButton__content"
        >
          <svg
            aria-hidden={true}
            className="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
            focusable="false"
            height={16}
            role="img"
            style={null}
            viewBox="0 0 16 16"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          />
          <span
            className="euiButton__text"
          >
            Save view
          </span>
        </span>
      </button>
    </div>
  </div>,
]
`;

exports[`Metrics Top Menu Component renders Top Menu Component when disabled with no metric visualizations 1`] = `
Array [
  <div
    className="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
  >
    <div
      className="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <div
        className="resolutionSelect"
      >
        <div
          className="euiFormControlLayout euiFormControlLayout--group"
        >
          <label
            className="euiFormLabel euiFormControlLayout__prepend"
          >
            Span Interval
          </label>
          <div
            className="euiFormControlLayout__childrenWrapper"
          >
            <input
              aria-label="resolutionField"
              className="euiFieldText resolutionSelectText euiFieldText--inGroup"
              data-test-subj="metrics__spanValue"
              disabled={true}
              onChange={[Function]}
              type="text"
              value={1}
            />
          </div>
          <div
            className="euiFormControlLayout"
          >
            <div
              className="euiFormControlLayout__childrenWrapper"
            >
              <select
                aria-label="resolutionSelect"
                className="euiSelect euiFormControlLayout__append resolutionSelectOption"
                data-test-subj="metrics__spanResolutionSelect"
                onChange={[Function]}
                onMouseUp={[Function]}
                value="h"
              >
                <option
                  key="0"
                  value="s"
                >
                  seconds
                </option>
                <option
                  key="1"
                  value="m"
                >
                  minutes
                </option>
                <option
                  key="2"
                  value="h"
                >
                  hours
                </option>
                <option
                  key="3"
                  value="d"
                >
                  days
                </option>
                <option
                  key="4"
                  value="y"
                >
                  years
                </option>
              </select>
              <div
                className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
              >
                <span
                  className="euiFormControlLayoutCustomIcon"
                >
                  <svg
                    aria-hidden={true}
                    className="euiIcon euiIcon--medium euiFormControlLayoutCustomIcon__icon"
                    focusable="false"
                    height={16}
                    role="img"
                    style={null}
                    viewBox="0 0 16 16"
                    width={16}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M13.069 5.157 8.384 9.768a.546.546 0 0 1-.768 0L2.93 5.158a.552.552 0 0 0-.771 0 .53.53 0 0 0 0 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 0 0 0-.76.552.552 0 0 0-.771 0Z"
                      fillRule="non-zero"
                    />
                  </svg>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="euiFlexItem metrics-search-bar-datepicker"
    >
      <div
        className="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--directionRow euiSuperDatePicker__flexWrapper"
      >
        <div
          className="euiFlexItem"
        >
          <div
            className="euiFormControlLayout euiFormControlLayout--group euiFormControlLayout-isDisabled euiSuperDatePicker"
          >
            <div
              className="euiPopover euiPopover--anchorDownLeft"
            >
              <div
                className="euiPopover__anchor euiQuickSelectPopover__anchor"
              >
                <button
                  aria-label="Date quick select"
                  className="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--xSmall euiButtonEmpty-isDisabled euiFormControlLayout__prepend"
                  data-test-subj="superDatePickerToggleQuickMenuButton"
                  disabled={true}
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    className="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                  >
                    <svg
                      aria-hidden={true}
                      className="euiIcon euiIcon--small euiIcon--inherit euiButtonContent__icon"
                      focusable="false"
                      height={16}
                      role="img"
                      style={null}
                      viewBox="0 0 16 16"
                      width={16}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M13.069 5.157 8.384 9.768a.546.546 0 0 1-.768 0L2.93 5.158a.552.552 0 0 0-.771 0 .53.53 0 0 0 0 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 0 0 0-.76.552.552 0 0 0-.771 0Z"
                        fillRule="non-zero"
                      />
                    </svg>
                    <span
                      className="euiButtonEmpty__text euiQuickSelectPopover__buttonText"
                    >
                      <svg
                        aria-hidden={true}
                        className="euiIcon euiIcon--medium"
                        focusable="false"
                        height={16}
                        role="img"
                        style={null}
                        viewBox="0 0 16 16"
                        width={16}
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M14 4v-.994C14 2.45 13.55 2 12.994 2H11v1h-1V2H6v1H5V2H3.006C2.45 2 2 2.45 2 3.006v9.988C2 13.55 2.45 14 3.006 14h9.988C13.55 14 14 13.55 14 12.994V5H2V4h12zm-3-3h1.994C14.102 1 15 1.897 15 3.006v9.988A2.005 2.005 0 0 1 12.994 15H3.006A2.005 2.005 0 0 1 1 12.994V3.006C1 1.898 1.897 1 3.006 1H5V0h1v1h4V0h1v1zM4 7h2v1H4V7zm3 0h2v1H7V7zm3 0h2v1h-2V7zM4 9h2v1H4V9zm3 0h2v1H7V9zm3 0h2v1h-2V9zm-6 2h2v1H4v-1zm3 0h2v1H7v-1zm3 0h2v1h-2v-1z"
                          fillRule="evenodd"
                        />
                      </svg>
                    </span>
                  </span>
                </button>
              </div>
            </div>
            <div
              className="euiFormControlLayout__childrenWrapper"
            >
              <div
                className="euiDatePickerRange euiDatePickerRange--inGroup"
              >
                <button
                  className="euiSuperDatePicker__prettyFormat euiSuperDatePicker__prettyFormat--disabled"
                  data-test-subj="superDatePickerShowDatesButton"
                  disabled={true}
                  onClick={[Function]}
                >
                  Last 1 day
                  <span
                    className="euiSuperDatePicker__prettyFormatLink"
                  >
                    Show dates
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <span
            className="euiToolTipAnchor"
            onKeyUp={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <button
              className="euiButton euiButton--primary euiButton--fill euiButton-isDisabled euiSuperUpdateButton"
              data-test-subj="superDatePickerApplyTimeButton"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "minWidth": undefined,
                }
              }
              type="button"
            >
              <span
                className="euiButtonContent euiButton__content"
              >
                <svg
                  aria-hidden={true}
                  className="euiIcon euiIcon--medium euiIcon--inherit euiButtonContent__icon"
                  focusable="false"
                  height={16}
                  role="img"
                  style={null}
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11.228 2.942a.5.5 0 1 1-.538.842A5 5 0 1 0 13 8a.5.5 0 1 1 1 0 6 6 0 1 1-2.772-5.058ZM14 1.5v3A1.5 1.5 0 0 1 12.5 6h-3a.5.5 0 0 1 0-1h3a.5.5 0 0 0 .5-.5v-3a.5.5 0 1 1 1 0Z"
                  />
                </svg>
                <span
                  className="euiButton__text euiSuperUpdateButton__text"
                >
                  Refresh
                </span>
              </span>
            </button>
          </span>
        </div>
      </div>
    </div>
    <div
      className="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <div
        className="euiPopover euiPopover--anchorDownCenter"
      >
        <div
          className="euiPopover__anchor"
        >
          <button
            className="euiButton euiButton--primary euiButton-isDisabled"
            data-test-subj="metrics__saveManagementPopover"
            disabled={true}
            onClick={[Function]}
            style={
              Object {
                "minWidth": undefined,
              }
            }
            type="button"
          >
            <span
              className="euiButtonContent euiButtonContent--iconRight euiButton__content"
            >
              <svg
                aria-hidden={true}
                className="euiIcon euiIcon--medium euiIcon--inherit euiButtonContent__icon"
                focusable="false"
                height={16}
                role="img"
                style={null}
                viewBox="0 0 16 16"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13.069 5.157 8.384 9.768a.546.546 0 0 1-.768 0L2.93 5.158a.552.552 0 0 0-.771 0 .53.53 0 0 0 0 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 0 0 0-.76.552.552 0 0 0-.771 0Z"
                  fillRule="non-zero"
                />
              </svg>
              <span
                className="euiButton__text"
              >
                Save
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="euiSpacer euiSpacer--m"
  />,
  <div
    className="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
  >
    <div
      className="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <button
        className="euiButton euiButton--primary euiButton--small euiButton-isDisabled"
        data-test-subj="metrics__editView"
        disabled={true}
        onClick={[Function]}
        style={
          Object {
            "minWidth": undefined,
          }
        }
        type="button"
      >
        <span
          className="euiButtonContent euiButton__content"
        >
          <svg
            aria-hidden={true}
            className="euiIcon euiIcon--medium euiIcon--inherit euiButtonContent__icon"
            focusable="false"
            height={16}
            role="img"
            style={null}
            viewBox="0 0 16 16"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.148 3.148 11 2l-9 9v3h3l9-9-1.144-1.144-8.002 7.998a.502.502 0 0 1-.708 0 .502.502 0 0 1 0-.708l8.002-7.998ZM11 1c.256 0 .512.098.707.293l3 3a.999.999 0 0 1 0 1.414l-9 9A.997.997 0 0 1 5 15H2a1 1 0 0 1-1-1v-3c0-.265.105-.52.293-.707l9-9A.997.997 0 0 1 11 1ZM5 14H2v-3l3 3Z"
            />
          </svg>
          <span
            className="euiButton__text"
          >
            Edit view
          </span>
        </span>
      </button>
    </div>
  </div>,
]
`;

exports[`Metrics Top Menu Component renders Top Menu Component when enabled 1`] = `
Array [
  <div
    className="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
  >
    <div
      className="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <div
        className="resolutionSelect"
      >
        <div
          className="euiFormControlLayout euiFormControlLayout--group"
        >
          <label
            className="euiFormLabel euiFormControlLayout__prepend"
          >
            Span Interval
          </label>
          <div
            className="euiFormControlLayout__childrenWrapper"
          >
            <input
              aria-label="resolutionField"
              className="euiFieldText resolutionSelectText euiFieldText--inGroup"
              data-test-subj="metrics__spanValue"
              disabled={false}
              onChange={[Function]}
              type="text"
              value={1}
            />
          </div>
          <div
            className="euiFormControlLayout"
          >
            <div
              className="euiFormControlLayout__childrenWrapper"
            >
              <select
                aria-label="resolutionSelect"
                className="euiSelect euiFormControlLayout__append resolutionSelectOption"
                data-test-subj="metrics__spanResolutionSelect"
                onChange={[Function]}
                onMouseUp={[Function]}
                value="h"
              >
                <option
                  key="0"
                  value="s"
                >
                  seconds
                </option>
                <option
                  key="1"
                  value="m"
                >
                  minutes
                </option>
                <option
                  key="2"
                  value="h"
                >
                  hours
                </option>
                <option
                  key="3"
                  value="d"
                >
                  days
                </option>
                <option
                  key="4"
                  value="y"
                >
                  years
                </option>
              </select>
              <div
                className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
              >
                <span
                  className="euiFormControlLayoutCustomIcon"
                >
                  <svg
                    aria-hidden={true}
                    className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                    focusable="false"
                    height={16}
                    role="img"
                    style={null}
                    viewBox="0 0 16 16"
                    width={16}
                    xmlns="http://www.w3.org/2000/svg"
                  />
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="euiFlexItem metrics-search-bar-datepicker"
    >
      <div
        className="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--directionRow euiSuperDatePicker__flexWrapper"
      >
        <div
          className="euiFlexItem"
        >
          <div
            className="euiFormControlLayout euiFormControlLayout--group euiSuperDatePicker"
          >
            <div
              className="euiPopover euiPopover--anchorDownLeft"
            >
              <div
                className="euiPopover__anchor euiQuickSelectPopover__anchor"
              >
                <button
                  aria-label="Date quick select"
                  className="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--xSmall euiFormControlLayout__prepend"
                  data-test-subj="superDatePickerToggleQuickMenuButton"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    className="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                  >
                    <svg
                      aria-hidden={true}
                      className="euiIcon euiIcon--small euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                      focusable="false"
                      height={16}
                      role="img"
                      style={null}
                      viewBox="0 0 16 16"
                      width={16}
                      xmlns="http://www.w3.org/2000/svg"
                    />
                    <span
                      className="euiButtonEmpty__text euiQuickSelectPopover__buttonText"
                    >
                      <svg
                        aria-hidden={true}
                        className="euiIcon euiIcon--medium euiIcon-isLoading"
                        focusable="false"
                        height={16}
                        role="img"
                        style={null}
                        viewBox="0 0 16 16"
                        width={16}
                        xmlns="http://www.w3.org/2000/svg"
                      />
                    </span>
                  </span>
                </button>
              </div>
            </div>
            <div
              className="euiFormControlLayout__childrenWrapper"
            >
              <div
                className="euiDatePickerRange euiDatePickerRange--inGroup"
              >
                <button
                  className="euiSuperDatePicker__prettyFormat"
                  data-test-subj="superDatePickerShowDatesButton"
                  disabled={false}
                  onClick={[Function]}
                >
                  Last 1 day
                  <span
                    className="euiSuperDatePicker__prettyFormatLink"
                  >
                    Show dates
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <span
            className="euiToolTipAnchor"
            onKeyUp={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <button
              className="euiButton euiButton--primary euiButton--fill euiSuperUpdateButton"
              data-test-subj="superDatePickerApplyTimeButton"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "minWidth": undefined,
                }
              }
              type="button"
            >
              <span
                className="euiButtonContent euiButton__content"
              >
                <svg
                  aria-hidden={true}
                  className="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                  focusable="false"
                  height={16}
                  role="img"
                  style={null}
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                />
                <span
                  className="euiButton__text euiSuperUpdateButton__text"
                >
                  Refresh
                </span>
              </span>
            </button>
          </span>
        </div>
      </div>
    </div>
    <div
      className="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <div
        className="euiPopover euiPopover--anchorDownCenter"
      >
        <div
          className="euiPopover__anchor"
        >
          <button
            className="euiButton euiButton--primary"
            data-test-subj="metrics__saveManagementPopover"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "minWidth": undefined,
              }
            }
            type="button"
          >
            <span
              className="euiButtonContent euiButtonContent--iconRight euiButton__content"
            >
              <svg
                aria-hidden={true}
                className="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                focusable="false"
                height={16}
                role="img"
                style={null}
                viewBox="0 0 16 16"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              />
              <span
                className="euiButton__text"
              >
                Save
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="euiSpacer euiSpacer--m"
  />,
  <div
    className="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
  >
    <div
      className="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <button
        className="euiButton euiButton--primary euiButton--small"
        data-test-subj="metrics__editView"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "minWidth": undefined,
          }
        }
        type="button"
      >
        <span
          className="euiButtonContent euiButton__content"
        >
          <svg
            aria-hidden={true}
            className="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
            focusable="false"
            height={16}
            role="img"
            style={null}
            viewBox="0 0 16 16"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          />
          <span
            className="euiButton__text"
          >
            Edit view
          </span>
        </span>
      </button>
    </div>
  </div>,
]
`;
